HOMEWORK_15/EX1.CPP

```diff
@@ -0,0 +1,240 @@
+#include <iostream>
+#include <algorithm>
+#include <string>
+#include <vector>
+
+struct Student
+{
+    std::string name;
+    int marks[4];
+
+    float averageMark;
+};
+
+void print(Student* students, int size)
+{
+    for(int i = 0; i < size; i++)
+    {
+        std::cout<<"Name: "<<students[i].name<<", Marks: ";
+        for(int j = 0; j < 4; j++)
+        {
+            std::cout<<students[i].marks[j]<<", ";
+        }
+        std::cout<<"Average mark: "<<students[i].averageMark<<std::endl;
+    }
+}
+
+float averageMark(Student* students, int number)
+{
+    float average = 0;
+    for(int i = 0; i < 4; i++)
+    {
+        average += students[number].marks[i];
+    }
+    average = average/4;
+    return average;
+}
+
+bool compare(const Student& student1, const Student student2)
+{
+    return student1.averageMark > student2.averageMark;
+}
+
+void sort(Student* students, int size)
+{
+    std::sort(students, students + size, compare);
+}
+
+Student* mostSuccessfulStudents(Student* students, int size)
+{
+    Student* mostSuccessful = &students[0];
+    for(int i = 1; i < size; i++)
+    {
+        if(students[i].averageMark > mostSuccessful->averageMark)
```

Comment:
if `averageMark` function was never called for a particular student then the `averageMark` field will be unset. That's is quite a big assumption you make here that someone called it

Reply:
Done